openpype/hosts/blender/scripts/build_workfile.py

```diff
@@ -648,10 +648,59 @@ def build_anim(project_name, asset_name):
     bpy.ops.scene.make_container_publishable(container_name=cam_container.name)
     cam_instance = bpy.context.scene.openpype_instances[-1]
 
+    animated_objects = set()
     for obj in bpy.context.scene.objects:
         if obj.type == "ARMATURE":
-            # Create animation instance
-            variant_name = obj.name[obj.name.find("RIG_") + 4 :].capitalize()
+            # Create animation instance for rig.
+            variant_name = (
+                obj.name[4].upper() + obj.name[5:]
+                if obj.name.find("RIG_") == 0
+                else obj.name[0].upper() + obj.name[1:]
+            )
+            bpy.ops.scene.create_openpype_instance(
+                creator_name="CreateAnimation",
+                asset_name=asset_name,
+                subset_name=f"animation{variant_name}",
+                datapath="objects",
+                datablock_name=obj.name,
+                use_selection=False,
+            )
+        elif obj.animation_data and obj.animation_data.action:
+            animated_objects.add(obj)
+
+    if animated_objects:
+        # Searching animated object from container.
+        for container in bpy.context.scene.openpype_containers:
+            # Get all objects from container.
+            container_objects = set()
+            for d_ref in container.datablock_refs:
+                if isinstance(d_ref.datablock, bpy.types.Collection):
+                    container_objects.update(d_ref.datablock.all_objects)
+                elif isinstance(d_ref.datablock, bpy.types.Object):
+                    container_objects.add(d_ref.datablock)
+            if not container_objects:
+                continue
+            # Get animation variant name using container short name.
+            trim_idx = container.name.find("_setdress")
+            variant_name = container.name[0].upper() + (
+                container.name[1:trim_idx]
+                if trim_idx > 0
+                else container.name[1:]
+            )
```

Comment:
Can't you use `rstrip`?

```suggestion
            variant_name = container.name.rstrip("_setdress").capitalize()
```

Reply:
```python
"KickAss_setdress".rstrip("_setdress").capitalize()  # Result: Kicka
```